### ISSUE TITLE
HttpContent not properly checked when selecting expected options

### ISSUE THREAD

**Issue by Kralizek** [PureKrome/HttpClient.Helpers#19]
Hi,

I think I found a bug in the FakeMessageHandler when it compares the incoming HttpContent and the expected ones received from the HttpMessageOptions.

The problem is caused by the fact that your equality check is a reference check.

```
private HttpMessageOptions GetExpectedOption(HttpMessageOptions option)
{
    if (option == null)
    {
        throw new ArgumentNullException(nameof(option));
    }

    IEnumerable<HttpMessageOptions> options = _lotsOfOptions.Values;

    options = options.Where(x => x.RequestUri == option.RequestUri || x.RequestUri == HttpMessageOptions.NoValue);

    options = options.Where(x => (x.HttpMethod == option.HttpMethod || x.HttpMethod == null));

    options = options.Where(x => (x.HttpContent == option.HttpContent || x.HttpContent == null));

    options = options.Where(x => (x.Headers == null || x.Headers.Count == 0) || (x.Headers != null && HeaderExists(x.Headers, option.Headers)));

    return options.SingleOrDefault();
}
```

If we look at this equivalent version of your code, the error is in the `options = options.Where(x => (x.HttpContent == option.HttpContent || x.HttpContent == null));` line.

I reproduced the issue by slightly changing the test. You can check it here: https://github.com/Kralizek/HttpClient.Helpers/commit/339ae338018eb10386c316a57a4b1e8ba606ba7f

**Comment by PureKrome** [comment]
Hi @Kralizek - thanks _heaps_ for looking into this library and possibly even using it :)

I'm literally just about to go to bed right now, so when I get up tomorrow, I'll review this issue.

Again - thanks heaps, mate!

**Comment by PureKrome** [comment]
OK - I agree that this is a bug :: it needs to check the _content_ of the `HttpContent`

While reviewing #18 I noticed that there's code in there that helps fix this but it looks pretty hardcore at stages .. like [this `GetHashCode`](https://github.com/PureKrome/HttpClient.Helpers/pull/18/files#diff-ee872b61ad765eb6c1cf997ad081b985R77) code.

So i'm wondering -> why don't we just do a check to see if the (`ReadAsByteArrayAsync`) `Content` of both source/destination are equiv instead of creating a full blown `IEqualityComparer` ?

**Comment by PureKrome** [comment]
OK . #25 includes a fix for this. So go and check that out and see if that looks ok to you.

I'm just doing a simple `string` vs `string` check.

NOTE: the check is a *case sensitive* string check.

**Comment by PureKrome** [comment]
Fixed in #25
